apps/aecore/src/aec_sync.erl

```diff
@@ -158,6 +163,23 @@ handle_call({next_work_item, STId, PeerId, LastResult}, _From, State) ->
     {reply, Reply, State2};
 handle_call({gossip_txs, GossipTxs}, _From, State) ->
     {reply, ok, State#state{ gossip_txs = GossipTxs }};
+handle_call(sync_progress, _, State) ->
+    State#state.sync_tasks,
+    SyncProgress =
+        case State#state.sync_tasks of
+            [] -> 100.0;
+            SyncTasks ->
+                TargetHeight =
+                    lists:foldl(
+                      fun(SyncTask, MaxHeight) ->
+                              #{chain := Chain} = SyncTask#sync_task.chain,
+                              [#{height := Height} | _] = Chain,
+                              max(Height, MaxHeight)
+                      end, 0, SyncTasks),
+                TopHeight = aec_headers:height(aec_chain:top_header()),
+                round(10000000 * TopHeight / TargetHeight) / 100000
```

Comment:
Why such hight precision here, I'd not expect more than double number precision to have any meaning

Reply:
It can be double precision if you prefer, we're very well prepared for long chains now (5 digits after `.`) 😉 . I don't care much 🙂

Reply:
No, I just recalled why double in not enough already (I remembered changing that, didn't remember why though 🙂 ):

```
➜  aeternity git:(pt-163476661-add-sync_progress-field-to-status-endpoint) curl 127.0.0.1:3013/v2/status
{"difficulty":46614936315406,"genesis_key_block_hash":"kh_pbtwgLrNu23k9PA6XCZnUbtsvEFeQGgavY4FS2do3QP8kcp2z","listening":true,"network_id":"ae_mainnet","node_revision":"0ea51ca9cfdd972506a656cd390f156d145f4bd0","node_version":"2.0.0","peer_count":390,"pending_transactions_count":222,"protocols":[{"effective_at_height":47800,"version":2},{"effective_at_height":0,"version":1}],"solutions":0,"sync_progress":95.68505,"syncing":true}%                                                                                                                                            ➜  aeternity git:(pt-163476661-add-sync_progress-field-to-status-endpoint) curl 127.0.0.1:3013/v2/status
{"difficulty":46614936315406,"genesis_key_block_hash":"kh_pbtwgLrNu23k9PA6XCZnUbtsvEFeQGgavY4FS2do3QP8kcp2z","listening":true,"network_id":"ae_mainnet","node_revision":"0ea51ca9cfdd972506a656cd390f156d145f4bd0","node_version":"2.0.0","peer_count":390,"pending_transactions_count":222,"protocols":[{"effective_at_height":47800,"version":2},{"effective_at_height":0,"version":1}],"solutions":0,"sync_progress":95.68718,"syncing":true}
```

Note `95.68505` vs `95.68718`.